### PR TITLE
upgrade dmidecode, dos2unix, dosfstools, gnutls, popt

### PR DESCRIPTION
**Pull request by nicogbg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

upgrade dmidecode, dos2unix, dosfstools, gnutls, popt

###### Change Log  

- relevant files

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/dmidecode/dmidecode.spec [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Summary:	Tool to analyze BIOS DMI data
 Name:		dmidecode
-Version:	3.2
-Release:        2%{?dist}
+Version:	3.3
+Release:    1%{?dist}
```
Comment (rlmenge) [inline SPECS/dmidecode/dmidecode.spec:4]: Nit: Spacing

File: SPECS/dmidecode/dmidecode.spec [hunk 2]
```diff
@@ -25,6 +25,10 @@ make DESTDIR=%{buildroot} prefix=%{_prefix} install
 %{_mandir}/man8/*
 
 %changelog
+* Fri Jan 21 2022 Nicolas Guibourge <nicolasg@microsoft.com> - 3.3-1
+- Upgrade to 3.3
+- License verified
+
 * Sat May 09 2020 Nick Samson <nisamson@microsoft.com> - 3.2-2
```
Comment (rlmenge) [inline SPECS/dmidecode/dmidecode.spec:32]: Nit: changelog spacing isn't uniform

File: SPECS/dosfstools/dosfstools.spec [hunk 3]
```diff
@@ -1,13 +1,17 @@
 Summary:	Dos Filesystem tools
 Name:		dosfstools
-Version:	4.1
-Release:        5%{?dist}
+Version:	4.2
+Release:    1%{?dist}
```
Comment (rlmenge) [inline SPECS/dosfstools/dosfstools.spec:4]: Nit: spacing

File: SPECS/dosfstools/dosfstools.spec [hunk 4]
```diff
@@ -31,6 +36,9 @@ make DESTDIR=%{buildroot} PREFIX="/usr" install
 %{_docdir}/dosfstools/*
 
 %changelog
+* Fri Jan 21 2022 Nicolas Guibourge <nicolasg@microsoft.com> - 4.2-1
```
Comment (rlmenge) [inline SPECS/dosfstools/dosfstools.spec:39]: Nit: changelog spacing isn't uniform

File: SPECS/dmidecode/dmidecode.spec [hunk 5]
```diff
@@ -1,7 +1,7 @@
 Summary:	Tool to analyze BIOS DMI data
 Name:		dmidecode
-Version:	3.2
-Release:        2%{?dist}
+Version:	3.3
+Release:    1%{?dist}
 License:	GPLv2+
 URL:		http://www.nongnu.org/dmidecode/
```
Comment (rlmenge) [inline SPECS/dmidecode/dmidecode.spec:6]: Nit: https?

File: SPECS/dosfstools/dosfstools.spec [hunk 6]
```diff
@@ -1,13 +1,17 @@
 Summary:	Dos Filesystem tools
 Name:		dosfstools
-Version:	4.1
-Release:        5%{?dist}
+Version:	4.2
+Release:    1%{?dist}
 License:	GPLv3+
 URL:		http://github.com/dosfstools/dosfstools
```
Comment (rlmenge) [inline SPECS/dosfstools/dosfstools.spec:6]: Nit: https?

File: SPECS/popt/popt.spec [hunk 7]
```diff
@@ -1,14 +1,13 @@
 Summary:	Programs to parse command-line options
 Name:		popt
-Version:	1.16
-Release:        7%{?dist}
+Version:	1.18
+Release:    1%{?dist}
```
Comment (rlmenge) [inline SPECS/popt/popt.spec:4]: Nit: spacing

File: SPECS/popt/popt.spec [hunk 8]
```diff
@@ -1,14 +1,13 @@
...
 License:	MIT
-URL:		http://rpm5.org/files/popt
+URL:		https://github.com/rpm-software-management/popt
 Group:		Applications/System
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
-Source0:	ftp://anduin.linuxfromscratch.org/BLFS/svn/p/%{name}-%{version}.tar.gz
-%define sha1 popt=cfe94a15a2404db85858a81ff8de27c8ff3e235e
+Source0:	http://ftp.rpm.org/popt/releases/popt-1.x/%{name}-%{version}.tar.gz
```
Comment (rlmenge) [inline SPECS/popt/popt.spec:10]: Nit: https

File: SPECS/popt/popt.spec [hunk 9]
```diff
@@ -60,6 +59,10 @@ make %{?_smp_mflags} check
 %defattr(-,root,root)
 
 %changelog
```
Comment (rlmenge) [inline SPECS/popt/popt.spec:61]: Nit: changelog spacing

**Review by rlmenge — APPROVED** [review]

File: SPECS/dosfstools/dosfstools.spec [hunk 10]
```diff
@@ -1,13 +1,17 @@
...
 Group:		Filesystem Tools
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
-Source0:	http://github.com/%{name}/%{name}/releases/download/v%{version}/%{name}-%{version}.tar.gz
+Source0:	https://github.com/dosfstools/dosfstools/archive/v4.2.tar.gz#/%{name}-%{version}.tar.gz
```
Comment (PawelWMS) [inline SPECS/dosfstools/dosfstools.spec:10]: Let's still use `%{version}` in the URL to make updates a bit easier:

```suggestion
Source0:	https://github.com/dosfstools/dosfstools/archive/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
```

**Review by PawelWMS — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

File: SPECS/dosfstools/dosfstools.spec (hunk 10 above)
Comment (nicogbg) [reply SPECS/dosfstools/dosfstools.spec:10]: good catch, thanks

File: SPECS/popt/popt.spec (hunk 8 above)
Comment (nicogbg) [reply SPECS/popt/popt.spec:10]: actually the https protocol is not supported by that server :(

Comment (nicogbg) [reply SPECS/popt/popt.spec:10]: and github release recommend to download source tarball from that http server :( :(

**Pull request merged**